tensorflow_probability/python/experimental/bijectors/highway_flow.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import tensorflow as tf
+
+import tensorflow_probability as tfp
+
+tfd = tfp.distributions
+tfb = tfp.bijectors
+tfe = tf.experimental
+
+'''
+This initial implementation is based on the PyTorch implementation at
+https://github.com/LucaAmbrogioni/CascadingFlow/blob/main/modules/networks.py
+'''
+
+
+class TriResNet(tfb.Bijector):
+
+    # set activation=False for last block
+    def __init__(self, width, activation=True, validate_args=False, name='tri_res_net'):
+        super(TriResNet, self).__init__(
+            validate_args=validate_args,
+            forward_min_event_ndims=0,
```

Comment:
Assuming that the values transformed by this bijector (arguments `x` and `y` to the forward and inverse methods, respectively) are intended to be vectors, then this should be `forward_min_event_ndims=1`. (otherwise the base class will screw up your log-det-Jacobian calculations by trying to sum them).